api/src/main/java/graphql/nadel/normalized/ValueToVariableValueCompiler.java

```diff
@@ -0,0 +1,107 @@
+package graphql.nadel.normalized;
+
+import graphql.language.ArrayValue;
+import graphql.language.BooleanValue;
+import graphql.language.FloatValue;
+import graphql.language.IntValue;
+import graphql.language.ObjectField;
+import graphql.language.ObjectValue;
+import graphql.language.StringValue;
+import graphql.language.TypeName;
+import graphql.language.Value;
+import graphql.language.VariableDefinition;
+import graphql.language.VariableReference;
+import graphql.normalized.NormalizedInputValue;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+public class ValueToVariableValueCompiler {
+
+    static VariableValueWithDefinition normalizedInputValueToVariable(NormalizedInputValue normalizedInputValue) {
+        Object variableValue = null;
+        Object normalizedInputValueValue = normalizedInputValue.getValue();
+        if (normalizedInputValueValue instanceof ObjectValue) {
+            variableValue = toVariableValue((ObjectValue) normalizedInputValueValue);
+        }
+        if (normalizedInputValueValue instanceof ArrayValue) {
+            variableValue = toVariableValues(((ArrayValue) normalizedInputValueValue).getValues());
+        }
+        if (normalizedInputValueValue instanceof List) {
```

Comment:
Man I hate how `normalizedInputValue.getValue` gives you an `Object` or and not a `Value`.

Do you need to handle `Map`s here?

Reply:
> Do you need to handle Maps here?
for args like this for example
```
type Mutation {
    foo1(arg1: JSON!, arg2: [JSON!]): String
}
```

`arg1` is going to be an `ObjectValue` and `arg2` is a `List`

similarly for input types
```
input InputWithJson {
    id: ID
    json: JSON
    jsons: [JSON]
}
```
`json` is an `ObjectValue` and jsons is a `List`